client/blocks/site-icon/index.jsx

```diff
@@ -36,7 +36,7 @@ function SiteIcon( { siteId, site, iconUrl, size, imgSize, isTransientIcon } ) {
 		<div className={ classes } style={ style }>
 			{ ! site && siteId > 0 && <QuerySites siteId={ siteId } /> }
 			{ iconSrc
-				? <Image className="site-icon__img" src={ iconSrc } />
+				? <Image className="site-icon__img" src={ iconSrc } alt="" />
```

Comment:
@ryelle I considered this but can't find anywhere that currently uses `SiteIcon` where that override would be needed, so would it be premature to provide that functionality?